frontend/src/citizen-frontend/header/DesktopNav.tsx

```diff
@@ -35,65 +36,72 @@ export default React.memo(function DesktopNav({
   unreadMessagesCount,
   unreadPedagogicalDocuments
 }: Props) {
-  const user = useUser()
+  const { user } = useContext(AuthContext)
   const t = useTranslation()
-
-  const isEnduser = user?.userType === 'ENDUSER'
-  const maybeLockElem = !isEnduser && (
-    <FontAwesomeIcon icon={faLockAlt} size="xs" />
-  )
   return (
-    <Container data-qa="desktop-nav">
-      <Nav>
-        {user && (
-          <>
-            <StyledNavLink to="/applying" data-qa="nav-applying">
-              {t.header.nav.applying} {maybeLockElem}
-            </StyledNavLink>
-            {user.accessibleFeatures.pedagogicalDocumentation && (
-              <StyledNavLink
-                to="/pedagogical-documents"
-                data-qa="nav-pedagogical-documents"
-              >
-                {t.header.nav.pedagogicalDocuments}
-                {isEnduser && unreadPedagogicalDocuments > 0 && (
-                  <CircledChar data-qa={'unread-pedagogical-documents-count'}>
-                    {unreadPedagogicalDocuments}
-                  </CircledChar>
-                )}
-                {maybeLockElem}
-              </StyledNavLink>
-            )}
-            {user.accessibleFeatures.messages && (
-              <StyledNavLink to="/messages" data-qa="nav-messages">
-                {t.header.nav.messages}{' '}
-                {unreadMessagesCount > 0 ? (
-                  <CircledChar>{unreadMessagesCount}</CircledChar>
-                ) : (
-                  ''
-                )}
-              </StyledNavLink>
-            )}
-            {user.accessibleFeatures.reservations && (
-              <StyledNavLink to="/calendar" data-qa="nav-calendar">
-                {t.header.nav.calendar}
-              </StyledNavLink>
+    <UnwrapResult result={user} loading={() => null}>
```

Comment:
Since `loading={() => null}` returns `null` when called the `<SpinnerSegment />` is rendered anyway. Could you fix that in `<UnwrapResult>`? 😄

Reply:
Oops :D That explains why there were two spinners in mobile widths. I thought it was an exisiting feature.